subsquid/src/bondedFinanceProcessor.ts

```diff
@@ -0,0 +1,103 @@
+import { EventHandlerContext } from "@subsquid/substrate-processor";
+import {
+  BondedFinanceNewBondEvent,
+  BondedFinanceNewOfferEvent,
+} from "./types/events";
+import { BondedFinanceBondOffer } from "./model";
+
+interface NewOfferEvent {
+  offerId: bigint;
+}
+
+interface NewBondEvent {
+  offerId: bigint;
+  nbOfBonds: bigint;
+}
+
+/**
+ * Extracts relevant information about a new bond
+ * @param event
+ */
+function getNewBondEvent(event: BondedFinanceNewBondEvent): NewBondEvent {
+  if (event.isV2300) {
+    const { offerId, nbOfBonds } = event.asV2300;
+    return { offerId, nbOfBonds };
+  }
+
+  const { offerId, nbOfBonds } = event.asLatest;
+  return { offerId, nbOfBonds };
+}
+
+function getNewOfferEvent(event: BondedFinanceNewOfferEvent): NewOfferEvent {
+  if (event.isV2300) {
+    const { offerId } = event.asV2300;
+
+    return { offerId };
+  }
+
+  const { offerId } = event.asLatest;
+  return { offerId };
+}
+
+/**
+ * Extracts beneficiary from a bond offer
+ * @param ctx
+ */
+function getBeneficiary(ctx: EventHandlerContext): string {
+  const { beneficiary } = ctx.event.extrinsic?.args[0]?.value as Record<
```

Comment:
Are we sure that this is the `beneficiary` always?

Reply:
Good question. It should, as `beneficiary` is indeed the first argument of the extrinsic.

Reply:
Hmm I don't think that is the case according to the rust code `BondOffer.beneficiary` can be set to something other than the first argument(`from`) of the extrinsic

Reply:
Ok, in that case I think the best way is to add it to the `NewOffer` event, which right now is only including the `offerId`. We could also obtain the beneficiary by querying the chain's state after we get the event. but that would be more error-prone in my opinion.

Reply:
> in that case I think the best way is to add it to the NewOffer event

Yeah that's the right approach

Reply:
Let's make a separate PR to make the rust updates.

Reply:
For reference: https://github.com/ComposableFi/composable/pull/1247